gef.py

```diff
@@ -2531,7 +2531,13 @@ def get_register(regname):
         value = gdb.parse_and_eval(regname)
         return to_unsigned_long(value) if value.type.code == gdb.TYPE_CODE_INT else long(value)
     except gdb.error:
-        value = gdb.selected_frame().read_register(regname)
+        assert(regname[0] == '$')
```

Comment:
We are generally not big fans of `assert` directive. Do you have a particular reason to use it except for trigger an exception? 

If not can you replace it with something simpler like:
```python
if regname[0]==0:
  do_stuff
else:
  return None
```

Reply:
I use `assert` because I don't think we will fail here ( I even think most people will not enter this `except`, like I described before ).
I can't image when this `assert` will fail, so I don't know how to deal with this situation.
Hope someone who fails into this `assert` will report us and we can handle with it.

Reply:
👍